### PR TITLE
DOCS-11236-db-xa-datasource-kt

### PR DESCRIPTION
**Pull request by kevintroller**
Adding new example for Datasource Reference Connection and XA mode

### CODE REVIEW

File: db/1.10/modules/ROOT/pages/database-connector-connection.adoc [hunk 1]
```diff
@@ -61,6 +61,84 @@ In the XML editor, both the spring and database configurations look like this:
 </db:config>
 ----
 
+== Configure a Data Source Reference Connection when in XA Transaction Mode
+
+If your Mule app uses Database Connector operations such as *Insert*, *Select* or others that run with an XA transaction, and connection pooling is enabled via the *Pooling profile* fields of the connector configuration, the default connection pool implementation for the Data Source Reference Connection is Enhydra XAPool.
```
Comment (nicomz) [inline db/1.10/modules/ROOT/pages/database-connector-connection.adoc]: I'd say "If your Mule App requires operations bound to XA transactions, and the connection pooling is enabled for the database configuration in use .."

**Review by juldiazmule — DISMISSED** [review]

File: db/1.10/modules/ROOT/pages/database-connector-connection.adoc [hunk 2]
```diff
@@ -61,6 +61,84 @@ In the XML editor, both the spring and database configurations look like this:
 </db:config>
 ----
 
+== Configure a Data Source Reference Connection when in XA Transaction Mode
```
Comment (kbostrom) [inline db/1.10/modules/ROOT/pages/database-connector-connection.adoc]:
```suggestion
== Configure a Data Source Reference Connection When in XA Transaction Mode
```

File: db/1.10/modules/ROOT/pages/database-connector-connection.adoc [hunk 3]
```diff
@@ -61,6 +61,84 @@ In the XML editor, both the spring and database configurations look like this:
...
+
+If your Mule app requires operations bound to XA transactions, and the connection pooling is enabled for the database configuration in use, the default connection pool implementation for the Data Source Reference Connection is Enhydra XAPool.
```
Comment (kbostrom) [inline db/1.10/modules/ROOT/pages/database-connector-connection.adoc]:
```suggestion
If your Mule app requires operations bound to XA transactions, and connection pooling is enabled for the database configuration in use, the default connection pool implementation for Data Source Reference Connection is Enhydra XAPool.
```

File: db/1.10/modules/ROOT/pages/database-connector-connection.adoc [hunk 4]
```diff
@@ -61,6 +61,84 @@ In the XML editor, both the spring and database configurations look like this:
...
+
+Note that Enhydra XAPool does not support prepared statements. To execute these operations use the Bitronix connection pool, refer to xref:mule-runtime::using-bitronix-to-manage-transactions.adoc#configuring-the-bitronix-transaction-manager[configuring the Bitronix Transaction Manager] documentation for details.
```
Comment (kbostrom) [inline db/1.10/modules/ROOT/pages/database-connector-connection.adoc]:
```suggestion
Note that Enhydra XAPool does not support prepared statements. To execute these operations, use the Bitronix connection pool. Refer to xref:mule-runtime::using-bitronix-to-manage-transactions.adoc#configuring-the-bitronix-transaction-manager[configuring the Bitronix Transaction Manager] documentation for details.
```

File: db/1.10/modules/ROOT/pages/database-connector-connection.adoc [hunk 5]
```diff
@@ -61,6 +61,84 @@ In the XML editor, both the spring and database configurations look like this:
...
+
+[NOTE]
```
Comment (kbostrom) [inline db/1.10/modules/ROOT/pages/database-connector-connection.adoc:70]: Is it odd to follow a paragraph beginning with "Note that..." with an actual [NOTE]-formatted comment?
Just asking; not sure, myself, one way or the other.

File: db/1.10/modules/ROOT/pages/database-connector-connection.adoc [hunk 6]
```diff
@@ -61,6 +61,84 @@ In the XML editor, both the spring and database configurations look like this:
...
+When using XA transactions, you must provide an `XADataSource` object in the JDBC data source spring bean configuration file of your app.
+
+The following example shows how to configure a data source reference connection for a Derby database when in XA transaction mode:
```
Comment (kbostrom) [inline db/1.10/modules/ROOT/pages/database-connector-connection.adoc]: Why is "a data source reference connection" lowercase here but not in Line 66? 
(Which is why I suggested removing "the" before the proper noun....)
Standardize.

File: db/1.10/modules/ROOT/pages/database-connector-connection.adoc [hunk 7]
```diff
@@ -61,6 +61,84 @@ In the XML editor, both the spring and database configurations look like this:
...
+
+. In Studio, in the `src/main/resources` folder, create a `datasource-config.xml` bean file.
```
Comment (kbostrom) [inline db/1.10/modules/ROOT/pages/database-connector-connection.adoc]:
```suggestion
. In Studio, in the `src/main/resources` folder, create a `datasource-config.xml` spring bean file.
```

File: db/1.10/modules/ROOT/pages/database-connector-connection.adoc [hunk 8]
```diff
@@ -61,6 +61,84 @@ In the XML editor, both the spring and database configurations look like this:
...
+. Add the `XADataSource` object at the end of the class `org.enhydra.jdbc.standard.Standard`, for example `class="org.enhydra.jdbc.standard.StandardXaDataSource`.
```
Comment (kbostrom) [inline db/1.10/modules/ROOT/pages/database-connector-connection.adoc]:
```suggestion
. Add the `XADataSource` object to the end of the class `org.enhydra.jdbc.standard.Standard`, for example, `class="org.enhydra.jdbc.standard.StandardXaDataSource`:
```

File: db/1.10/modules/ROOT/pages/database-connector-connection.adoc [hunk 9]
```diff
@@ -61,6 +61,84 @@ In the XML editor, both the spring and database configurations look like this:
...
+
+    <bean id="jdbcDataSource" class="org.enhydra.jdbc.standard.StandardXaDataSource" destroy-method="shutdown">
+        <property name="driverName" value="org.apache.derby.jdbc.EmbeddedDriver"/>
+        <property name="url" value="jdbc:derby:muleEmbeddedDB;create=true"/>
+    </bean>
+
+</beans>
+----
++
+. Return to your Mule flow, go to *Global Elements* > *Create* to open the *Choose Global Type* window.
```
Comment (kbostrom) [inline db/1.10/modules/ROOT/pages/database-connector-connection.adoc]:
```suggestion
. Return to your Mule flow and navigate to *Global Elements* > *Create* to open the *Choose Global Type* window.
```

File: db/1.10/modules/ROOT/pages/database-connector-connection.adoc [hunk 10]
```diff
@@ -61,6 +61,84 @@ In the XML editor, both the spring and database configurations look like this:
...
+. In *Filter*, type `spring`, select *Spring Config*, and click *OK*.
+. In *Spring Config* > *Files*, add `datasource-config.xml` to reference the spring configuration file with the Spring module.
```
Comment (kbostrom) [inline db/1.10/modules/ROOT/pages/database-connector-connection.adoc:97]: "spring" > "spring bean"?
You know best; please just search for and check that _spring, bean_, and _spring bean_ are used consistently in this topic set.

File: db/1.10/modules/ROOT/pages/database-connector-connection.adoc [hunk 11]
```diff
@@ -61,6 +61,84 @@ In the XML editor, both the spring and database configurations look like this:
...
+. In *Filter*, type `database`, select *Database Config*, and click *OK*.
+. In *Database Config* > *Connection*, select *Data Source Reference Connection*.
+. Click *Configure* to add the required JDBC driver and select either: +
++
+* *Use local file* +
+Install the library using a local file.
+* *Add Maven dependency* +
+Install a Maven dependency to add to the project.
++
+. For *Data source ref*, add the bean id `jdbcDataSource` created in the `datasource-config.xml` bean file.
```
Comment (kbostrom) [inline db/1.10/modules/ROOT/pages/database-connector-connection.adoc]:
```suggestion
. For *Data source ref*, add the bean ID `jdbcDataSource` created in the `datasource-config.xml` bean file.
```

File: db/1.10/modules/ROOT/pages/database-connector-connection.adoc [hunk 12]
```diff
@@ -61,6 +61,84 @@ In the XML editor, both the spring and database configurations look like this:
...
+. In the *Advanced* tab, configure the *Pooling profile*, *Column types* and *Reconnection* fields.
```
Comment (kbostrom) [inline db/1.10/modules/ROOT/pages/database-connector-connection.adoc]:
```suggestion
. In the *Advanced* tab, configure the *Pooling profile*, *Column types*, and *Reconnection* fields.
```

**Review by kbostrom — DISMISSED** [review]
CE

File: db/1.10/modules/ROOT/pages/database-connector-connection.adoc [hunk 13]
```diff
@@ -61,6 +61,84 @@ In the XML editor, both the spring and database configurations look like this:
 </db:config>
 ----
 
+== Configure a Data Source Reference Connection When in XA Transaction Mode
+
+If your Mule app requires operations bound to XA transactions, and connection pooling is enabled for the database configuration in use, the default connection pool implementation for Data Source Reference Connection is Enhydra XAPool.
+
+Enhydra XAPool does not support prepared statements. To execute these operations use the Bitronix connection pool. Refer to xref:mule-runtime::using-bitronix-to-manage-transactions.adoc#configuring-the-bitronix-transaction-manager[configuring the Bitronix Transaction Manager] documentation for details.
```
Comment (dhirsch1) [inline db/1.10/modules/ROOT/pages/database-connector-connection.adoc]: The first word in the link should be capitalized - Configuring

**Review by dhirsch1 — APPROVED** [review]

**Pull request merged**